src/list.js

```diff
@@ -0,0 +1,101 @@
+export default class List {
+  constructor() {
+    const currentData = JSON.parse(localStorage.getItem('todo-list'));
+    if (currentData) {
+      this.items = currentData;
+    } else {
+      this.items = [
+        {
+          description: 'Play Games At Liquid Telecoms Hackathon',
+          completed: false,
+          index: 1,
+        },
+        {
+          description: 'Study JavaScript At 2:45 PM',
+          completed: false,
+          index: 2,
+        },
+        {
+          description: 'Talk To Girlfriend',
+          completed: false,
+          index: 3,
+        },
+        {
+          description: 'High Customer Success Manager',
+          completed: false,
+          index: 4,
+        },
+      ];
+    }
+  }
+
+  displayItems() {
+    this.saveItems();
+    const listSection = document.querySelector('.items');
+    listSection.innerHTML = '';
+    this.items.forEach((task) => {
+      let listItem = `
+    <li>`;
+      if (task.completed === true) {
+        listItem += `
+      <span class="material-icons update-status done" data-id="${task.index}">done</span>
+      <p class="line-through activity" data-id="${task.index}"  contenteditable="true">${task.description}</p>`;
+      } else {
+        listItem += `
+      <span class="material-icons update-status" data-id="${task.index}">check_box_outline_blank</span>
+      <p  class="activity" data-id="${task.index}" contenteditable="true">${task.description}</p>`;
+      }
+      listItem += `
+    <span class="material-icons more-info">more_vert</span>
+    </li>`;
+      listSection.innerHTML += listItem;
+    });
+    this.attachEvents();
+  }
+
+  updateStatus(index) {
+    const arrIndex = index - 1;
+    const currentStatus = this.items[arrIndex].completed;
+    if (currentStatus === true) {
+      currentStatus.completed = false;
+    } else {
+      currentStatus.completed = true;
+    }
+    this.displayItems();
+  }
+
+  saveItems() {
+    // reorder indexes
+    this.items.forEach((task, index) => {
+      this.items[index].index = index + 1;
+    });
+    // sort indexes
+    this.items.sort((a, b) => {
+      if (a.index < b.index) return -1;
+      if (a.index > b.index) return 1;
+      return 0;
+    });
+    // save sorted items to localStorage
+    localStorage.setItem('todo-list', JSON.stringify(this.items));
+  }
+
+  attachEvents() {
+    const tasks = document.querySelectorAll('.update-status');
+    tasks.forEach((task) => {
+      task.addEventListener('click', (e) => {
```

Comment:
- [ ] The task's checkbox should have the [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) event listener as this is part of the requirements. Kindly fix that

Reply:
Hi @EricMbouwe,

Thank you for the feedback, I'm working to fix that now except it is bringing out a few issues, I'll try to restructure my design slightly. Thank you once more- I really appreciate the good review.

Reply:
@EricMbouwe 

Just a quick follow-up, please. I tried using the `change` event but then I released it doesn't work because I'm using icons for `check box` and the `blue check`, however, the design still achieves the same thing. If I'm to use the `change` event, I'd have to use default styling of the `input type="checkbox"` which will be different from the design. Kindly advise, please.

Reply:
Hi @JEFFKAY16
Thanks for reaching out about that. You just have to use the `input` element of `type=checkbox` to handle that. Don't worry too much about the design while it does the job that is expected. One of the main objectives of this project is to work with the `change` event listener to create interactive effect

Reply:
This issue is well resolved! ✔️